### DISCUSSION TITLE
Transcript to Subtitles

### DISCUSSION THREAD

**Discussion by ngozz** [SubtitleEdit/subtitleedit#7650] [Q&A] [answered]
As title, I have a video and a transcript of it. The video is quite long so I want to find a way to automated the process of adding timestamps to the transcript and turn it into subtitles. I believe currently SE can't do this.

Any help/idea/suggestion would be appreciated. Thank you.

Update: I tried the program suggested by ext666 and it worked really well. Just one small problem, that project wasn't made to work with big files (long videos, subtitles and transcripts). I had to split the files in order for it to work. Hope there will be a better solution in the future!

**Comment by darnn** [comment]
https://github.com/SubtitleEdit/subtitleedit/issues/7623#issuecomment-1807940348

**Reply by ngozz** [reply]
Ty. I got an approach, would you mind giving me some advice? Using SE Audio to Text Whisper, I can get a fairly good subtitles. I'll try to write a script to compare this subtitles to the transcript somehow and get the correct lines.

I'm not familiar with WhisperX so it could take some time. I'll try it after experimenting with some stuff.

**Reply by darnn** [reply]
I mean, that sounds plausible enough, but as someone who has no experience with coding at all, I don't know how easy it would be to actually achieve. Like I said, I'd try looking into wav2vec first, to use that to match your audio to your transcript that has no timecodes. If you're going to use Whisper to generate timecodes at all, I'd do so with WhisperX, which uses wav2vec, since in my tests (not exhaustive, but still) that gave timecodes that were closest to the real ones.

If you want to mess around with WhisperX yourself but are having a hard time getting it to run on your machine (like I am), you could do it through Kaggle:
https://www.kaggle.com/

Register, create a new notebook, upload the file you want to transcribe and then just do the following:
`!pip install git+https://github.com/m-bain/whisperx.git`
`!whisperx --language en --model large --compute_type float32 "/kaggle/input/dataa/filename.mp3"`
Where "dataa" is the "dataset title" you enter when you upload a file, and "filename.mp3" is, naturally, the file you've uploaded.

**Reply by ngozz** [reply]
That's a very thorough answer. Thank you.

**Comment by darnn** [comment]
No problem! As you can tell, I wish this was possible myself. If you make any progress on it, please keep us posted!

**Comment by ext666** [comment, marked as answer]
> I'll try to write a script to compare this subtitles to the transcript somehow and get the correct lines.

That script already exists. I use it and it works really well.
https://github.com/EtienneAb3d/WhisperTimeSync

**Reply by ngozz** [reply]
Omg, tysm. I'll try it out.
